web/src/app/admin/skills/skills.component.ts

```diff
@@ -1,14 +1,105 @@
-import {Component, OnInit} from '@angular/core';
+import { Component, OnInit } from '@angular/core';
+import { SkillsService } from './shared/skills.serviceApp';
+import { SkillViewModel } from '../../webapi/models/skill-view-model';
 
 @Component({
   selector: 'skills',
-  templateUrl: 'skills.component.html'
+  templateUrl: 'skills.component.html',
+  styleUrls: ['./skills.component.css']
 })
 
 export class SkillsComponent implements OnInit {
-  constructor() {
-  }
 
+  skillsModel: SkillViewModel = {} as SkillViewModel;
+  skills: SkillViewModel[] = [];
+  skillsList: SkillViewModel[];
+
+  isCreate: boolean = false;
+  isUpdate: boolean = false;
+  isUserExists: boolean = false;
+
+  addskills(skillsList: SkillViewModel) {
+    this.skillsList.push(skillsList)
+  }
+  constructor(private SkillsServiceApp: SkillsService) {
+  }
   ngOnInit() {
+    this.listSkill();
+  }
+  model: any = {};
+  msg: any = "";
+
+  onSubmit(skillsform) {
+    if (skillsform.valid) {
+      if (this.skillsModel['skillId'] === undefined) {
+        this.SkillsServiceApp.addSkill(this.skillsModel).subscribe(
+          (data) => {
+            console.log(data.body);
+            this.skills.push(data.body);
+            console.log("skills", this.skills)
+            this.msg = " created successfully";
+          }
+        );
+        this.skillsModel.name = "";
+        this.skillsModel.description = "";
+      } else {
+        this.SkillsServiceApp.updateSkill(this.skillsModel).subscribe(
+          (data) => {
+            this.listSkill();
+            this.msg = " updated successfully";
+          }
+        );
+      }
+      console.log(name, this.skillsModel.name);
+      if (name == this.skillsModel.name) {
+      }
+      document.getElementById('updatePanel').style.display = 'block';
+      document.getElementById('btnUpdate').style.display = 'inline';
+      this.skillsModel.name = "";
+      this.skillsModel.description = "";
+    }
+    this.isCreate = true;
+  }
+  checkNameExitsOnBlur() {
+    let $this = this;
+    let skillName = $this.skillsModel.name;
+    this.skills.forEach(function (skill) {
+      if (skill['name'] === skillName) {
+        $this.isUserExists = true;
+        return;
+      }
+    });
+  }
+  listSkill() {
+    this.SkillsServiceApp.listSkill()
+      .subscribe((data) => {
+        if (data) {
+          this.skills = data.body;
+          console.log(this.skills)
+        }
+      })
+  }
+  deleteSkills(i) {
+    this.skills.splice(i, 1);
```

Comment:
use db call to remove the skill; also need msg prompt before deletion

Reply:
done